src/util/hash.rs

```diff
@@ -21,8 +21,39 @@ use util::pair::*;
 
 use crypto::ripemd160::Ripemd160;
 use crypto::sha2::Sha256;
+use crypto::sha3::Sha3;
 use crypto::digest::Digest;
 
+#[derive(Serialize, Deserialize)]
+pub struct Keccak256Hash(pub [u8; 32]);
+impl Keccak256Hash {
```

Comment:
If we're going to expose sha256 to smart contracts, why is keccak256 needed?

Reply:
Same answer to previous keccak256 comment (providing easier interop capabilities with common systems using keccak256).